perf/src/bin/perf_server.rs

```diff
@@ -53,7 +53,7 @@ async fn main() {
 
 async fn run(opt: Opt) -> Result<()> {
     let (key, cert) = match (&opt.key, &opt.cert) {
-        (&Some(ref key), &Some(ref cert)) => {
+        (Some(key), Some(cert)) => {
```

Comment:
@Ralith I remember you have opinions on this but they don't align with my own. I think you care more than I do so while this looks like a good change to me, happy to defer to your judgement here.

Reply:
I'm generally not a fan of this lint (https://github.com/rust-lang/rust-clippy/issues/10245) because it obfuscates whether you're borrowing or moving out of the match scrutinee. I'm not going to die on this hill, though.